src/server/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp

```diff
@@ -503,6 +529,31 @@ class boss_shade_of_aran : public CreatureScript
                     return;
                 }
         }
+
+        void MoveInLineOfSight(Unit* who) override
+        {
+            ScriptedAI::MoveInLineOfSight(who);
+
+            if (SeenAtiesh || me->IsInCombat() || me->GetDistance2d(who) > me->GetAttackDistance(who) + 10.0f)
+                return;
+
+            Player* player = who->ToPlayer();
+            if (!player)
+                return;
+
+            for (auto id : AtieshStaves)
```

Comment:
true, but in source has more cases like: ```for (uint8 i = 0; i < 4; ++i)```
i think Jackpoz that complain about this "auto", idk what is real difference